src/main/java/org/openhds/service/impl/census/IndividualService.java

```diff
@@ -39,9 +71,45 @@ public Individual recordIndividual(Individual individual,
                                        String membershipUuid,
                                        String relationshipUuid,
                                        String fieldWorkerId) {
+
         individual.setCollectedBy(fieldWorkerService.findOrMakePlaceHolder(fieldWorkerId));
 
-        //TODO: Handle side effect creation for things like socialgroup and membership etc.
+        //Create the individual first, so that everything after is guaranteed to point at a real entity
+        individual = createOrUpdate(individual);
+
+        //TODO: Is the socialgroup created here?
+        //TODO: Do we make two paths depending on if it's a member or head of household?
+        SocialGroup socialGroup = socialGroupService.findOrMakePlaceHolder(socialGroupUuid);
+        Location location = locationService.findOrMakePlaceHolder(locationUuid);
+
+        //TODO: How are we going to start using project codes?
+        //TODO: Do these side effect entities need to call findOrMakePlaceholder?
+        /* It's my understanding that these entities are only ever created when an individual is created
+         this means that a placeholder doesn't exist yet. However, there is really no harm in calling
+         the method and overwriting whatever it returns, even if it's NOT a placeholder (e.g. some other
+         implementation supports the explicit creation of things like memberships outside of the context of
+         creating an individual... Should we be checking if the entity is return is a placeholder or not? if so, how?
+         There is currently no normalized way of checking all entities 'types' (placeholder, unknown, normal, etc). */
+        Residency residency = residencyService.findOrMakePlaceHolder(residencyUuid);
+        residency.setStartDate(ZonedDateTime.now());
+        residency.setStartType("PROJECT-CODE-GOES-HERE");
+        residency.setIndividual(individual);
+        residency.setLocation(location);
+        residencyService.createOrUpdate(residency);
+
+        Membership membership = membershipService.findOrMakePlaceHolder(membershipUuid);
+        membership.setStartDate(ZonedDateTime.now());
+        membership.setStartType("PROJECT-CODE-GOES-HERE");
+        membership.setIndividual(individual);
+        membership.setSocialGroup(socialGroup);
+        membershipService.createOrUpdate(membership);
+        //TODO: Is relationship to group head still a valid field of a membership?
+
+
+        Relationship relationship = relationshipService.findOrMakePlaceHolder(relationshipUuid);
+        //TODO: what relationship do we create here? Relationship to head of household?
```

Comment:
This seems like a use case question for Bruce.  What flavors of "record individual" does OpenHDS support?

Once we know the answer from a domain point of view, we can figure out which ids the caller needs to pass in, like maybe headOfHouseholdUuid.